cleanlab/internal/outlier.py

```diff
@@ -70,6 +70,6 @@ def transform_distances_to_scores(
     ood_features_scores: np.ndarray = np.exp(-1 * avg_distances / scaling_factor * t)
 
     # Set scores to 1 if the average distance is close to 0
-    inds = np.isclose(avg_distances, 0)
+    inds = np.isclose(avg_distances, 0, atol=5e-6)
```

Comment:
I don't think we should be changing our source code to fix unit tests. Should instead fix the unit test. This atol value was added here for a reason, but will let @elisno share how to best proceed

Reply:
It turns out tthat he default `atol` was a bit too small for the euclidean distance, so we explicitly set the `atol` parameter based how large the floating-point error can get for the distance computations.

It's still not 100% verified how this happens on only a few platforms for the euclidean distances, but we're confident that we can put a MUCH lower tolerance for the cosine metric.
The takeaway is: **A single `atol` value won't work** for different types of datasets, we need to set it carefully when comparing values against 0.

Reply:
It's not within the scope of this PR, but we may want to look at similar things for the near-duplicate check.